### PR TITLE
feat: Implements support for OPC UA for GMS

### PR DESCRIPTION
**Pull request by Kantiran91**
* Adds GMS Release Candidate support
* Implements BasicGMS
* Implements OGP and Wenzel machine instances

### CODE REVIEW

File: TypeDefinition/GMS/GMSEquipmentType.hpp [hunk 1]
```diff
@@ -0,0 +1,25 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/.
+ *
+ * Copyright 2020-2021 (c) Christian von Arnim, ISW University of Stuttgart (for umati and VDW e.V.)
+ * Copyright 2020 (c) Tonja Heinemann, ISW University of Stuttgart (for umati and VDW e.V.)
+ */
```
Comment (GoetzGoerisch) [inline TypeDefinition/GMS/GMSEquipmentType.hpp:7]: Add copyright note.

File: TypeDefinition/MachineryResult/Constants.hpp [hunk 2]
```diff
@@ -0,0 +1,14 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/.
+ *
+ * Copyright 2020-2021 (c) Christian von Arnim, ISW University of Stuttgart (for umati and VDW e.V.)
+ */
```
Comment (GoetzGoerisch) [inline TypeDefinition/MachineryResult/Constants.hpp:6]: Add copyright note.

File: model/GMS/Opc.Ua.MachineryResult.NodeSet2.xml [hunk 3]
```diff
@@ -0,0 +1,2488 @@
+<?xml version="1.0" encoding="utf-8"?>
```
Comment (GoetzGoerisch) [inline model/GMS/Opc.Ua.MachineryResult.NodeSet2.xml:1]: Add license header

File: model/GMS/opc.ua.gms.nodeset2.xml [hunk 4]
```diff
@@ -0,0 +1,3315 @@
+<?xml version="1.0" encoding="utf-8"?>
```
Comment (GoetzGoerisch) [inline model/GMS/opc.ua.gms.nodeset2.xml:1]: Add license header

File: CMakeLists.txt [hunk 5]
```diff
@@ -181,6 +181,48 @@ target_sources(
 )
 target_link_libraries(ua_gen_woodworking PUBLIC ua_gen_machinery)
 
+ua_generate_nodeset_and_datatypes(
```
Comment (GoetzGoerisch) [inline CMakeLists.txt:184]: Add copyright note to header

**Review by GoetzGoerisch — CHANGES_REQUESTED** [review]

File: TypeDefinition/MachineryResult/Constants.hpp (hunk 2 above)
Comment (Kantiran91) [reply TypeDefinition/MachineryResult/Constants.hpp:7]: done

File: CMakeLists.txt (hunk 5 above)
Comment (Kantiran91) [reply CMakeLists.txt:185]: done

File: TypeDefinition/GMS/GMSEquipmentType.hpp (hunk 1 above)
Comment (Kantiran91) [reply TypeDefinition/GMS/GMSEquipmentType.hpp:7]: done

File: model/GMS/opc.ua.gms.nodeset2.xml (hunk 4 above)
Comment (Kantiran91) [reply model/GMS/opc.ua.gms.nodeset2.xml:1]: done

File: model/GMS/Opc.Ua.MachineryResult.NodeSet2.xml (hunk 3 above)
Comment (Kantiran91) [reply model/GMS/Opc.Ua.MachineryResult.NodeSet2.xml:1]: done

**Review by GoetzGoerisch — APPROVED** [review]

**Comment by GoetzGoerisch** [comment]
Waiting for the upstream fix of deps/open62541 to be merged.

**Comment by GoetzGoerisch** [comment]
@Kantiran91 @karstenroethig  upstream fix is included into `develop` branch.
Rebased and cleanup this branch. Please give it a review before we merge?

**Comment by karstenroethig** [comment]
Looks good. Tested with newest Docker image.

I just added [adba8bf](https://github.com/umati/Sample-Server/pull/659/commits/adba8bf5f9dbdabe115a5ecc689ac0f4683ec4b5) to append the os name to the uploaded artifacts

**Review by karstenroethig — APPROVED** [review]

**Pull request merged**